src/components/Form.js

```diff
@@ -0,0 +1,217 @@
+        this.props.onSubmit(values);
+    }
+
+    /**
+     * @param {Object} values - An object containing the value of each inputID
+     * @returns {Object} - An object containing the errors for each inputID
+     */
+    validate(values) {
+        FormActions.setServerErrorMessage(this.props.formID, '');
+        const validationErrors = this.props.validate(values);
+        const errors = _.pick(validationErrors, (inputValue, inputID) => (
+            Boolean(this.touchedInputs[inputID])
+        ));
+        this.setState({errors});
+        return errors;
+    }
+
+    render() {
+        const childrenWrapperWithProps = children => (
+            // eslint-disable-next-line rulesdir/prefer-underscore-method
+            React.Children.map(children, (child) => {
+                // Just render the child if it is not a valid React element, e.g. text within a <Text> component
+                if (!React.isValidElement(child)) {
+                    return child;
+                }
+
+                // Depth first traversal of the render tree as the input element is likely to be the last node
+                if (child.props.children) {
+                    return React.cloneElement(child, {
+                        children: childrenWrapperWithProps(child.props.children),
+                    });
+                }
+
+                // We check if the child has the isFormInput prop.
+                // We don't want to pass form props to non form components, e.g. View, Text, etc
+                if (!child.props.isFormInput) {
+                    return child;
+                }
+
+                // We clone the child passing down all form props
+                const inputID = child.props.inputID;
+                return React.cloneElement(child, {
+                    ref: node => this.inputRefs[inputID] = node,
+                    defaultValue: this.props.draftValues[inputID] || child.props.defaultValue,
+                    errorText: this.state.errors[inputID] || '',
+                    onBlur: (event) => {
+                        this.setTouchedInput(inputID);
+                        this.validate(this.getValues());
+                        if (child.props.onBlur) {
+                            child.props.onBlur(event);
+                        }
+                    },
+                    onChange: (value) => {
+                        // Expose value to the input ref so we can access it on native too
+                        this.inputRefs[inputID].value = value;
+
+                        if (child.props.shouldSaveDraft) {
+                            FormActions.setDraftValues(this.props.formID, {[inputID]: value});
+                        }
+                        if (this.touchedInputs[inputID]) {
+                            this.validate(this.getValues());
+                        }
+                        if (child.props.onChange) {
+                            child.props.onChange(value);
```

Comment:
I agree that it might encourage inconsistencies and I added it to handle side effects as you mentioned. The only example that comes to mind is possibly `NewChatPage`, but I have to take a closer look at it. That being said, I don't feel strongly about keeping it here. I'll remove it for now and we can add this back if we have a use case for it.